File: 2015/Day 01/day01.cpp
//
//  Advent of Code 2015
//  Day 01
//
//  By PatchesTheDipstick
//


//  System includes...


//  Third Party includes...



//  Project includes...

#include "../utils.hpp"

int main(int argc, char** argv)
{
	
	std::vector<std::string> input = getInput("input.txt");
	
	int current_floor = 0;
	int first_enter_basement = -1;
	int instruction_counter = 0;
	
	for (int a = 0; a < input.size(); a++)
	{
	
		for (int b = 0; b < input[a].length(); b++)
		{
			if (input[a][b] == '(')
			{
				current_floor++;
				instruction_counter++;
			}
			else
			{
				current_floor--;
				instruction_counter++;
				if ((current_floor < 0) && (first_enter_basement == -1))
				{
					first_enter_basement = instruction_counter;
				}
			}
		}
	}

		
	int answer_1 = current_floor;
	int answer_2 = first_enter_basement;
	
	std::cout << "AoC 2015 Day 1 answers:" << std::endl;
	std::cout << "    Part 1 : " << answer_1 << std::endl;
	std::cout << "    Part 2 : " << answer_2 << std::endl;

	return (0);

}

File: 2015/Day 01/day01.rs
use std::env;
use std::fs;

fn main()
{
    let contents = fs::read_to_string("input.txt")
	    .expect("Something F'ed up!");
		
	let mut current_floor = 0;
	let mut when_underground = 0;
	let mut count = 0;
	for c in contents.chars()
	{
		count = count + 1;
		if c == '(' 
		{
			current_floor = current_floor + 1;
		}
		if c == ')'
		{
			current_floor = current_floor - 1;
			if current_floor < 0 && when_underground == 0
			{
				when_underground = count;
			}
		}
	}
	println!("AoC 2015 Day 01 :");
	println!("  Part 1 : {}", current_floor);
	println!("  Part 2 : {}", when_underground);
}